crates/bevy_asset/src/path.rs

```diff
@@ -200,3 +245,23 @@ impl<'a> From<String> for AssetPath<'a> {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_relative_path() {
```

Comment:
Awesome that you added tests! Would you be able to add a few more? Namely, ones that should fail/panic, as well as some more unusual paths (like the multiple `./` one I mentioned in a previous comment).

Reply:
This whole file needs tests, actually. For example, there really should be tests for constructing an AssetPath from a string.

Reply:
Done.